provider/mesh/state.go

```diff
@@ -0,0 +1,350 @@
+package mesh
+
+import (
+	"bytes"
+	"encoding/gob"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/prometheus/alertmanager/provider"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/satori/go.uuid"
+	"github.com/weaveworks/mesh"
+)
+
+func utcNow() time.Time { return time.Now().UTC() }
+
+type notificationEntry struct {
+	Resolved  bool
+	Timestamp time.Time
+}
+
+type notificationState struct {
```

Comment:
Doc comment: notificationState implements mesh.GossipData.
(Those things are really helpful when reading the code. Go-oracle can tell me the same thing, but it takes long enough that I wouldn't hit the hotkey for each struct I see.)